### PR TITLE
automatic NOFILE increase, show help at no arg, ...

### LINKED ISSUE #3: `--recursive` option is being ignored, remove.
I'm willing to send a patch for this issue by myself, and I'm going to ask before I do.

What do you think? Even if we implement this option, I think it would be more usable to leave recursive as a default behavior and add a `--no-recursive` option.

### LINKED ISSUE #4: Print help message when no argument is given
I'm willing to send a patch for this issue by myself, and I'm going to ask before I do.

What do you think?

### PR DESCRIPTION
**Pull request by simnalamburt**
Behavioral changes:

1. [Automatically increase NOFILE rlimit to the allowed maximum](https://github.com/cr0sh/jaso/commit/e73aecd4c99c7babdd1419182742dd3ebeb96961)
2. [Print help message when no argument is given, add --version option](https://github.com/cr0sh/jaso/commit/ab5c830041c3947ae4a092f230b1f479f5d75223)
3. [Remove unused --recursive option](https://github.com/cr0sh/jaso/commit/85f101cb1e45982740078f797e15356399e1ad47)

Non-behavioral changes:

1. Added CI for tests, clippy, fmt
2. Fixed all clippy warnings (removed unused imports)
3. etc

Closes #3
Closes #4

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -10,10 +10,8 @@ const CONCURRENT_TASKS: usize = 32768;
 static SEMA: Semaphore = Semaphore::const_new(CONCURRENT_TASKS);
 
 #[derive(Parser, Debug)]
-#[clap(rename_all = "kebab-case")]
```
Comment (cr0sh) [inline src/main.rs:13]: I'm curious why this is removed, is it a default behavior?

File: src/main.rs [hunk 2]
```diff
@@ -28,14 +26,17 @@ struct Args {
 #[tokio::main]
 async fn main() {
     let args = Args::parse();
-    let nofile = rlimit::getrlimit(Resource::NOFILE).expect("cannot query rlimit");
 
+    // Automatically increase NOFILE rlimit to the allowed maximum
+    rlimit::increase_nofile_limit(u64::MAX).expect("failed during increasing NOFILE rlimit");
```
Comment (cr0sh) [inline src/main.rs:31]: Didn't know that this is supported. Great job!

File: src/main.rs (hunk 1 above)
Comment (simnalamburt) [reply src/main.rs:13]: Yeap https://github.com/cr0sh/jaso/pull/5/commits/6de48769c8622b7553830008830bf5b1d5348525

###### Reference
- https://docs.rs/clap/latest/clap/_derive/index.html#command-attributes

**Comment by cr0sh** [comment]
Oops forgot to approve(anyway the merge implies that xD)

**Pull request merged**